### PR TITLE
Add functions spm unit and build tests

### PR DESCRIPTION
**Pull request by jrcrespoh**
Tests ran successfully with watchos

`pod lib lint` fails on travis for watchos, but succeeds locally. Disabling test for now, pending GHA update to Xcode 12.5+

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -690,6 +690,24 @@ let package = Package(
...
     ),
+    .testTarget(
+      name: "FunctionsUnit",
+      dependencies: ["FirebaseFunctions",
+                     "SharedTestUtilities"],
+      path: "Functions/Example/Tests/",
+      exclude: [
+        "en.lproj",
+        "Tests-Info.plist",
+      ],
```
Comment (jrcrespoh) [inline Package.swift:701]: This was added because otherwise it would complain about not having a `defaultLocalization` for spm.

**Review by ncooke3 — COMMENTED** [review]
Nice! I think we should add `watchos` to the each matrix in `functions.yml` [Here's one such matrix.](https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/functions.yml#L27)

**Comment by paulb777** [comment]
Looks like `pod lib lint` does not work for watch OS. https://travis-ci.org/github/firebase/firebase-ios-sdk/jobs/774699071

File: Functions/Tests/Unit/Swift/FunctionsAPITests.swift [hunk 2]
```diff
@@ -61,9 +61,9 @@ final class FunctionsAPITests {
     }
 
     #if swift(>=5.5)
-      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *) {
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *) {
```
[the remark this answers is not in the document]
Comment (ncooke3) [reply Functions/Tests/Unit/Swift/FunctionsAPITests.swift:64]: cc: @granluo This is a compile time test

File: Package.swift (hunk 1 above)
Comment (ncooke3) [reply Package.swift:701]: @paulb777 Have you seen this issue before?

**Review by ncooke3 — APPROVED** [review]
This LGTM 👍 . Added one comment about SwiftPM excludes as I havent seen that issue before

File: Package.swift (hunk 1 above)
Comment (paulb777) [reply Package.swift:701]: Good catch @ncooke3! I think those are files that are no longer used and should be deleted.  I'll investigate.

**Comment by jrcrespoh** [comment]
What should be done about incremental code coverage reports? I imagine these should not surface on tests?

**Comment by paulb777** [comment]
@granluo is working on the incremental coverage reporting separately.

@jrcrespoh This PR should be good to resolve and merge here.

**Pull request merged**